From 962dc32431b710b7db864e6ea1bbafaa3c9df251 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Petr=20P=C3=ADsa=C5=99?= <ppisar@redhat.com>
Date: Wed, 21 Jan 2015 10:36:54 +0100
Subject: [PATCH] Skip pax extended headers
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

This is a Archive-Tar commit:

commit 381e47992fced3e7b8a53a7be4739b7cf6f01719
Author: Chris Williams <chris@bingosnet.co.uk>
Date:   Fri Jan 7 22:38:54 2011 +0000

    Skip pax extended headers

    git-svn-id: http://oss.dwim.org/archive-tar-new@2861 4dccba1d-3c1b-0410-aec4-feb514a8dabc

ported to perl-5.10.1.

Signed-off-by: Petr Písař <ppisar@redhat.com>
---
 lib/Archive/Tar.pm | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/Archive/Tar.pm b/lib/Archive/Tar.pm
index 29c2a86..74d260f 100644
--- a/lib/Archive/Tar.pm
+++ b/lib/Archive/Tar.pm
@@ -451,7 +451,7 @@ sub _read_tar {
         ### skip this entry if it's a pax header. This is a special file added
         ### by, among others, git-generated tarballs. It holds comments and is
         ### not meant for extracting. See #38932: pax_global_header extracted
-        } elsif ( $entry->name eq PAX_HEADER ) {
+        } elsif ( $entry->name eq PAX_HEADER or $entry->type =~ /^(x|g)$/ ) {
             next LOOP;
         }
 
-- 
1.9.3

